### PR TITLE
add leetcode Permutation in String

### PR DESCRIPTION
**Pull request by alexpantyukhin**
#### Description of Change

Add a guide for new LeetCode solutions.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [ ] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: leetcode/src/567.c [hunk 1]
```diff
@@ -0,0 +1,66 @@
+const int EnglishLettersNumber = 26;
+
+void countCharsForStringSlice(int* charsCounter, char* s, int length, int sign){
```
Comment (StepfenShawn) [inline leetcode/src/567.c:3]:
```suggestion
void countCharsForStringSlice(int* charsCounter, char* s, int length, int sign) {

```

File: leetcode/src/567.c [hunk 2]
```diff
@@ -0,0 +1,66 @@
+const int EnglishLettersNumber = 26;
+
+void countCharsForStringSlice(int* charsCounter, char* s, int length, int sign){
+    for(int i = 0; i < length; i++){
```
Comment (StepfenShawn) [inline leetcode/src/567.c:4]:
```suggestion
    for (int i = 0; i < length; i++) {

```

File: leetcode/src/567.c [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        charsCounter[s[i] - 'a'] += sign;
+    }
+}
+
+
+// Sliding window
+// Calculate number of chars in the current slide. 
+// Runtime: O(n)
+// Space: O(1) - only number of english lowercase letters.
+bool checkInclusion(char* s1, char* s2){
```
Comment (StepfenShawn) [inline leetcode/src/567.c:14]:
```suggestion
bool checkInclusion(char* s1, char* s2) {

```

File: leetcode/src/567.c [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    int lengthS1 = strlen(s1);
+    int lengthS2 = strlen(s2);
+
+    if (lengthS1 > lengthS2){
```
Comment (StepfenShawn) [inline leetcode/src/567.c:18]:
```suggestion
    if (lengthS1 > lengthS2) {

```

File: leetcode/src/567.c [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    int* charsCounter = calloc(EnglishLettersNumber, sizeof(int));
+
+    // We keep counters of s1 with '-' sign. It has to be offset by s2 chars
+    countCharsForStringSlice(charsCounter, s1, lengthS1, -1);
+    countCharsForStringSlice(charsCounter, s2, lengthS1, 1);
+
+    int diffChars = 0;
+    for(int i = 0; i < EnglishLettersNumber; i++){
```
Comment (StepfenShawn) [inline leetcode/src/567.c:29]:
```suggestion
    for (int i = 0; i < EnglishLettersNumber; i++) {

```

File: leetcode/src/567.c [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        if (charsCounter[i] != 0) {
+            diffChars++;
+        }
+    }
+
+    if (diffChars == 0){
```
Comment (StepfenShawn) [inline leetcode/src/567.c:35]:
```suggestion
    if (diffChars == 0) {

```

File: leetcode/src/567.c [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        return true;
+    }
+
+    for(int i = 0; i < lengthS2 - lengthS1; i++){
```
Comment (StepfenShawn) [inline leetcode/src/567.c:39]:
```suggestion
    for (int i = 0; i < lengthS2 - lengthS1; i++) {

```

File: leetcode/src/567.c [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        int charNumberLeft = s2[i] - 'a';
+        int charNumberRight = s2[i + lengthS1] - 'a'; 
+
+        charsCounter[charNumberLeft] -= 1;
+        if (charsCounter[charNumberLeft] == 0){
```
Comment (StepfenShawn) [inline leetcode/src/567.c:44]:
```suggestion
        if (charsCounter[charNumberLeft] == 0) {

```

File: leetcode/src/567.c [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+            diffChars -= 1;
+        }
+        else if (charsCounter[charNumberLeft] == -1){
```
Comment (StepfenShawn) [inline leetcode/src/567.c:47]:
```suggestion
        else if (charsCounter[charNumberLeft] == -1) {

```

File: leetcode/src/567.c [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+            diffChars += 1;
+        }
+
+        charsCounter[charNumberRight] += 1;
+        if (charsCounter[charNumberRight] == 0){
```
Comment (StepfenShawn) [inline leetcode/src/567.c:52]:
```suggestion
        if (charsCounter[charNumberRight] == 0) {

```

File: leetcode/src/567.c [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+            diffChars -= 1;
+        }
+        else if (charsCounter[charNumberRight] == 1){
```
Comment (StepfenShawn) [inline leetcode/src/567.c:55]:
```suggestion
        else if (charsCounter[charNumberRight] == 1) {

```

File: leetcode/src/567.c [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+            diffChars += 1;
+        }
+
+        if (diffChars == 0){
```
Comment (StepfenShawn) [inline leetcode/src/567.c:59]:
```suggestion
        if (diffChars == 0) {

```

**Review by StepfenShawn — CHANGES_REQUESTED** [review]
Just a few changes about code format. :)

**Comment by alexpantyukhin** [comment]
@StepfenShawn Fixed! thank you!

**Review by Panquesito7 — DISMISSED** [review]
Thanks! 🚀

**Review by StepfenShawn — DISMISSED** [review]

**Review by tjgurwara99 — APPROVED** [review]
LGTM. There are quite a few extra vertical line spacing, is there a reason why you added them?

**Comment by alexpantyukhin** [comment]
> LGTM. There are quite a few extra vertical line spacing, is there a reason why you added them?

do you mean between methods? Actually no, it's by mistake. I will fix.

**Comment by alexpantyukhin** [comment]
> LGTM. There are quite a few extra vertical line spacing, is there a reason why you added them? It makes reading code difficult. If it was to keep logically separate blocks separate, then at least the first lines shouldn't have any whitespace.
> 
> So instead of this
> 
> ```
> int something() {
> 
>   int ss = 1;
>   do_something_else(ss);
> 
>   int s = 2;
>   go_for_it(s);
> }
> ```
> 
> we would have
> 
> ```
> int something() {
>   int ss = 1;
>   do_something_else(ss);
> 
>   int s = 2;
>   go_for_it(s);
> }
> ```
> 
> ^ makes it easy to read on smaller screens (IMO).
> 
> The whitespace in first line is redundant IMO but I've still approved it since this is a simple matter of opinion.

Fixed!:)

**Review by tjgurwara99 — APPROVED** [review]

**Review by StepfenShawn — APPROVED** [review]

**Pull request merged**